mountpoint-s3/src/fs.rs

```diff
@@ -1057,7 +1057,10 @@ where
 
     pub async fn unlink(&self, parent_ino: InodeNo, name: &OsStr) -> Result<(), Error> {
         if !self.config.allow_delete {
-            return Err(err!(libc::EPERM, "deletes are disabled"));
+            return Err(err!(
+                libc::EPERM,
+                "Deletes are disabled. Use '--allow-delete' mount option to enable it. \nFor more information on deletion, visit: https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes"
```

Comment:
I'm kind of not sure if we want to add documentation links in the logs. It's probably hard to maintain between different versions. As you mentioned, it could be broken if we move the docs to somewhere else or it's probably not true anymore when we decided to change the semantics.